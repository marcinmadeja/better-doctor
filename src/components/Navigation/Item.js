// @flow strict
import * as React from 'react';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';

const StyledIcon = styled.span`
  display: flex;
  align-items: center;
  height: 100%;
`;

type Props = {
  icon: React.Node,
  children: React.Node,
  path: string,
  className?: string,
};

const Item = ({ icon, path, children, className }: Props) => {
  const c_className: string = className || '';
  const componentClasses = classnames({ [c_className]: className });

  return (
    <MenuItem
      to={path}
      exact
      component={NavLink}
      className={componentClasses}
    >
      <ListItemIcon>
        <StyledIcon>
          {icon}
        </StyledIcon>
      </ListItemIcon>

      {children}
    </MenuItem>
  );
};

Item.defaultProps = {
  className: '',
};

export default Item;
