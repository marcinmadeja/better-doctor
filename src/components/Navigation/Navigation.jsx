// @flow strict
import * as React from 'react';
import MenuList from '@material-ui/core/MenuList';
import Item from './Item';

type Props = {
  children: React.ChildrenArray<React.Element<typeof Item>>
};

class Navigation extends React.Component<Props> {
  static Item = Item;

  render() {
    const { children } = this.props;

    return (
      <MenuList>
        {children}
      </MenuList>
    );
  }
}

export default Navigation;
