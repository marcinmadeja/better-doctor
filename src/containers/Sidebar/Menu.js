// @flow strict
import * as React from 'react';
import type { Route } from 'routes/routes';
import { withStyles } from '@material-ui/core';
import { THEME_COLORS } from 'constants/index';
import Navigation from 'components/Navigation/Navigation';

type Props = {
  list: Array<Route>,
  classes: { [string]: string }
}

const styles = {
  navItem: {
    color: THEME_COLORS.darkText,
    '& svg': {
      width: 22,
      color: THEME_COLORS.darkText,
    },

    '&.active': {
      color: THEME_COLORS.darkContrast,
      '& svg': {
        color: THEME_COLORS.darkContrast,
      },
    },
  },
};

export const Menu = ({ list, classes }: Props) => {
  return (
    <Navigation>
      {list.map(route => (
        <Navigation.Item
          key={route.path}
          icon={route.icon}
          path={route.path}
          className={classes.navItem}
        >
          {route.name}
        </Navigation.Item>
      ))}
    </Navigation>
  );
};

export default withStyles(styles)(Menu);
