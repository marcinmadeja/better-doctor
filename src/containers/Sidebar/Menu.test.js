import React from 'react';
import { MemoryRouter } from 'react-router';
import Navigation from 'components/Navigation/Navigation';
import { Menu } from './Menu';

const defaultProps = { classes: {}, list: [] };
const List = [
  {
    exact: true,
    path: '/',
    component: () => <div />,
    name: 'Home',
    icon: <span />,
  },
  {
    exact: true,
    path: '/test',
    component: () => <div />,
    name: 'Home 2',
    icon: <span data-testid="icon" />,
  },
];

const setup = (props = {}, initialEntries = '/') => {
  props = { ...defaultProps, ...props };
  const actions = {};
  const component = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Menu {...actions} {...props} />
    </MemoryRouter>
  );
  const shallowComponent = shallow(component);

  return {
    actions,
    component,
    shallowComponent,
  };
};

describe('Menu', () => {
  it('should render', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });

  it('should render correctly when proper List passed', () => {
    const { component } = setup({ list: List });

    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should add active class to path "/test"', () => {
    const testRoute = List[1];
    const { component } = setup({ list: List }, testRoute.path);
    const renderComponent = mount(component);
    const activeLink = renderComponent.find(Navigation.Item).last().find('.active').last();
    expect(activeLink.length).toEqual(1);
    expect(activeLink.text()).toEqual(testRoute.name);
  });
});
