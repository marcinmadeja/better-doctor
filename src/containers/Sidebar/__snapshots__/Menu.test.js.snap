// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly when proper List passed 1`] = `
<div
  className="MuiList-root-1 MuiList-padding-2"
  onBlur={[Function]}
  onKeyDown={[Function]}
  role="menu"
>
  <a
    aria-current="page"
    className="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiMenuItem-root-5 active"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    style={Object {}}
    tabIndex={-1}
  >
    <span
      className="MuiListItemIcon-root-20 sc-bdVaJa gguFeu"
    >
      <span />
    </span>
    Home
    <span
      className="MuiTouchRipple-root-21"
    />
  </a>
  <a
    aria-current={null}
    className="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiMenuItem-root-5"
    href="/test"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    style={undefined}
    tabIndex={-1}
  >
    <span
      className="MuiListItemIcon-root-20 sc-bdVaJa gguFeu"
    >
      <span
        data-testid="icon"
      />
    </span>
    Home 2
    <span
      className="MuiTouchRipple-root-21"
    />
  </a>
</div>
`;
